modules/test/playwright/tests/object-web/objectActions.spec.ts

```diff
@@ -71,17 +85,185 @@ test.describe('manage object actions through object actions tab', () => {
+
+	const objectDefinition = await apiHelpers.objectAdmin.postObjectDefinition({
+		active: true,
+		externalReferenceCode: 'customObjectERC',
+		label: {
+			en_US: 'customobject',
+		},
+		name: 'CustomObject',
+		objectFields: [
+			{
+				DBType: 'Long',
+				businessType: 'Attachment',
+				externalReferenceCode: 'testAttachment',
+				indexed: true,
+				indexedAsKeyword: false,
+				indexedLanguageId: '',
+				label: {
+					en_US: 'testAttachment',
+				},
+				listTypeDefinitionId: 0,
+				name: 'testAttachment',
+				objectFieldSettings: [
+					{
+						name: 'acceptedFileExtensions',
+						value: 'jpeg, jpg, pdf, png, txt',
+					},
+					{
+						name: 'maximumFileSize',
+						value: 100,
+					},
+					{
+						name: 'fileSource',
+						value: 'userComputer',
+					},
+					{
+						name: 'showFilesInDocumentsAndMedia',
+						value: false,
+					},
+				],
+				required: false,
+				state: false,
+				system: false,
+				type: 'Long',
+				unique: false,
+			},
+		],
+		pluralLabel: {
+			en_US: 'customobjects',
+		},
+		portlet: true,
+		scope: 'company',
+		status: {
+			code: 0,
+		},
+	});
+
+	createdEntities.objectDefinitonIds = [objectDefinition.id];
+
+	// Create an action to send notification after attachment download
+
+	await apiHelpers.objectAdmin.postObjectActionByExternalReferenceCode(
+		objectDefinition.externalReferenceCode,
+		{
+			active: true,
+			label: {
+				en_US: 'downloadAttachmentArchive',
+			},
+			name: 'downloadAttachmentArchive',
+			objectActionExecutorKey: 'notification',
+			objectActionTriggerKey: 'onAfterAttachmentDownload',
+			parameters: {
+				notificationTemplateId: notificationTemplate.id,
+				type: 'email',
+			},
+		}
+	);
+
+	// Create an object entry
+
+	await viewObjectEntriesPage.goto(objectDefinition.id);
+
+	await viewObjectEntriesPage.clickAddObjectEntry();
+
+	const fileChooserPromise = page.waitForEvent('filechooser');
+
+	await viewObjectEntriesPage.selectFileButton.click();
+
+	const fileChooser = await fileChooserPromise;
+
+	await fileChooser.setFiles(
+		path.join(__dirname, 'dependencies', 'sampleFile.txt')
+	);
+
+	await viewObjectEntriesPage.page
+		.getByText('sampleFile.txt')
+		.waitFor({state: 'visible'});
+
+	await viewObjectEntriesPage.saveObjectEntryButton.click();
+
+	await waitForSuccessAlert(page);
+
+	// Download attachment from object entry
+
+	await viewObjectEntriesPage.goto(objectDefinition.id);
+
+	await page
+		.getByRole('button', {name: 'Search'})
+		.waitFor({state: 'visible'});
+
+	await viewObjectEntriesPage.page.getByText('sampleFile.txt').click();
+
+	// Verify if the email was sent
+
+	const notificationQueueEntries =
+		await apiHelpers.notification.getNotificationQueueEntriesPage(
+			'do-not-replay@liferay.com'
+		);
+
+	createdEntities.notificationQueueEntryIds =
+		notificationQueueEntries.items.map((item: any) => item.id);
+
+	expect(notificationQueueEntries.items.length).toBeTruthy();
```

Comment:
This assertion is not good. What if there is like ANY notification from `do-not-replay@liferay.com` from another test that was not properly cleaned up? We would have this passing. We need to assert if our specific notification is in the queue. 

One wait that I found was this: 

`expect(notificationQueueEntries.items.some((item) => item.recipientsSummary === notificationTemplate.recipients[0].to.en_US)).toBeTruthy();
`

it does the job but it looks ugly! I am sure you are going to find a more elegant way! :heart: